### PR TITLE
Allow for use of stronger AES and SHA crypto

### PR DESCRIPTION
**Pull request by marri5317**
Update switch-cases to allow the use of stronger crypto pairs supported by GoSNMP

### CODE REVIEW

**Comment by brian-brazil** [comment]
CI doesn't like this, also the generate readme will need updating.

@SuperQ Now that there's more of these, gosnmp should probably expose this as a map or something so we don't have to maintain this here.

File: generator/README.md [hunk 1]
```diff
@@ -78,9 +78,9 @@ modules:
                                     # Can be noAuthNoPriv, authNoPriv or authPriv.
       password: pass  # Has no default. Also known as authKey, -A option to NetSNMP.
                       # Required if security_level is authNoPriv or authPriv.
-      auth_protocol: MD5  # MD5 or SHA, defaults to MD5. -a option to NetSNMP.
+      auth_protocol: MD5  # MD5, SHA, SHA224, SHA256, SHA384 or SHA512 defaults to MD5. -a option to NetSNMP.
```
Comment (SuperQ) [inline generator/README.md:81]:
```suggestion
      auth_protocol: MD5  # MD5, SHA, SHA224, SHA256, SHA384, or SHA512. Defaults to MD5. -a option to NetSNMP.
```

File: generator/README.md [hunk 2]
```diff
@@ -78,9 +78,9 @@ modules:
...
                           # Used if security_level is authNoPriv or authPriv.
-      priv_protocol: DES  # DES or AES, defaults to DES. -x option to NetSNMP.
+      priv_protocol: DES  # DES, AES, AES192 or AES256 defaults to DES. -x option to NetSNMP.
```
Comment (SuperQ) [inline generator/README.md:83]:
```suggestion
      priv_protocol: DES  # DES, AES, AES192, or AES256. Defaults to DES. -x option to NetSNMP.
```

**Comment by SuperQ** [comment]
Can you rebase your branch against master? That should fix the tests.

**Comment by SuperQ** [comment]
Ugh, every other CI run flip-flops the generator output.

**Comment by SuperQ** [comment]
Please rebase in order to pick up some fixes to the generator.

**Comment by marri5317** [comment]
Should be good to merge now

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**